### PR TITLE
Add a `HEAD` for metadata to verify dataset deleted

### PR DESCRIPTION
**Pull request by portante**
Possible use of a `HEAD` method for checking the presence of a dataset.

### CODE REVIEW

File: lib/pbench/test/functional/server/test_put.py [hunk 1]
```diff
@@ -142,6 +199,40 @@ def test_index_all(self, server_client: PbenchServerClient, login_user):
...
+            resource_id = datasets_hash.get(t.name)
+            assert resource_id, f"Expected to find tar ball {t.name} to delete"
+            response = server_client.post(
+                api=API.DATASETS_DELETE,
+                uri_params={"dataset": resource_id},
+                raise_error=False,
+            )
+            assert response.ok, f"{response.text}"
+            print(f"Deleted {t.name}")
+            response = server_client.head(
```
Comment (ndokos) [inline lib/pbench/test/functional/server/test_put.py:234]: This fails with

```
"HEAD /api/v1/datasets/metadata/22a4bc5748b920c6ce271eb68f08d91c HTTP/1.1" 405 0
```
I guess this was what was being discussed at the backlog grooming: `HEAD` is not allowed?

**Review by ndokos — CHANGES_REQUESTED** [review]

File: lib/pbench/test/functional/server/test_put.py (hunk 1 above)
Comment (portante) [reply lib/pbench/test/functional/server/test_put.py:234]: This is a `Draft` PR and this change is not correct.

Comment (ndokos) [reply lib/pbench/test/functional/server/test_put.py:234]: So given #3146, this should be closed?

Comment (portante) [reply lib/pbench/test/functional/server/test_put.py:234]: Once it is rebased, there is other work that this relies on from other PRs which may prove useful.

I am not going to do anything with this PR for the time being.

**Comment by portante** [comment]
No longer working on pbench.

**Pull request closed without merging**